app/src/main/java/lt/vilnius/tvarkau/API/LegacyApiModule.java

```diff
@@ -0,0 +1,96 @@
+package lt.vilnius.tvarkau.API;
+
+import com.facebook.stetho.okhttp3.StethoInterceptor;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.format.DateTimeFormatter;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import lt.vilnius.tvarkau.network.TokenAuthenticator;
+import lt.vilnius.tvarkau.network.TokenInterceptor;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+@Module
+public class LegacyApiModule {
+
+    public static final String API_BASE_URL = "http://www.vilnius.lt/m/m_problems/files/mobile/";
+
+    @Provides
+    @Singleton
+    public OkHttpClient provideOkHttpClient() {
+
+        return new OkHttpClient.Builder()
+            .authenticator(new TokenAuthenticator())
+            .addNetworkInterceptor(new Interceptor() {
+                @Override public Response intercept(Chain chain) throws IOException {
+                    Request.Builder requestBuilder = chain.request().newBuilder();
+                    requestBuilder.header("Content-Type", "application/json");
+                    return chain.proceed(requestBuilder.build());
+                }
+            })
+            .addNetworkInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
```

Comment:
Logging should be disabled in production. There's flag `BuildConfig.DEBUG` - could turn on logging only when it is `true`